AddonConfiguration well documented in kyma-project.io (#4758)
kyma-project/kyma: docs/helm-broker/06-01-clusteraddonsconfiguration.md

---
title: ClusterAddonsConfiguration
type: Custom Resource
---

The `clusteraddonsconfiguration.addons.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to define cluster-wide addons fetched by the Helm Broker. To get the up-to-date CRD and show the output in the `yaml` format, run this command:

```
kubectl get crd clusteraddonsconfiguration.addons.kyma-project.io -o yaml
```

## Sample custom resource

> This is a sample ClusterAddonsConfiguration which provides cluster-wide addons. If any status of ClusterAddonsConfiguration is marked as `Failed`, all of its addons are not available in the Service Catalog.

Reviewer: ```suggestion
This is a sample ClusterAddonsConfiguration which provides cluster-wide addons. If any of the **status** fields of the CR is marked as `Failed`, none of the addons registered with the CR are available in the Service Catalog.
```

> >**NOTE:** All CRs must have the `addons.kyma-project.io` finalizer which prevents the CR from deletion until the Controller completes the deletion logic successfully. If you do not set a finalizer, the Controller sets it automatically.

Reviewer: ```suggestion
>**NOTE:** All CRs must have the `addons.kyma-project.io` finalizer which prevents the CR from deletion until the Controller completes the deletion logic successfully. If you don't set a finalizer, the Controller sets it automatically.
```

          message: "An addon with the same ID is already registered: [ConfigurationName: addons-cfg, url: https://github.com/kyma-project/bundles/releases/download/0.4.0/index.yaml, addons: aws-service-broker:0.0.1]"
        - name: azure-service-broker
          version: 0.0.1
          status: Ready
    - url: https://github.com/kyma-project/bundles/releases/download/0.6.0/index-testing.yaml
      status: Ready
      addons:
        - name: testing
          version: 0.0.1
          status: Failed
          reason: ConflictInSpecifiedRepositories
          message: "Specified repositories have addons with the same ID: [url: https://github.com/kyma-project/bundles/releases/download/0.6.0/index.yaml, addons: gcp-service-broker:0.0.2]"
        - name: redis
          version: 0.0.3
          status: Failed
          reason: ValidationError
          message: "Addon validation failed due to error: schema /plans/default/update-instance-schema.json is larger than 64 kB"
    - url: https://broker.url
      status: Failed
      reason: FetchingIndexError
      message: "Fetching repository failed due to error: the index file was not found"
```

> >**NOTE:** There is no fast return in case of an error, which means that the Controller fetches and processes all addons, even if any of them fails. Thanks to that, at the end of the process you can see the status of all processed addons. You can read information about all detected problems in the **status** entry of a given CR.

Reviewer: ```suggestion
>**NOTE:** The Controller fetches and processes all addons, even if any of them fails. Thanks to that, at the end of the process you can see the status of all processed addons. You can read information about all detected problems in the **status** entry of a given CR.
```
